txscript/scriptnum.go

```diff
@@ -165,10 +164,10 @@ func (n scriptNum) Int32() int32 {
 // requireMinimal enabled.
 //
 // See the Bytes function documentation for example encodings.
-func makeScriptNum(v []byte, requireMinimal bool) (scriptNum, error) {
-	// Interpreting data as an integer requires that it is not larger than
-	// a 32-bit integer.
-	if len(v) > maxScriptNumLen {
+func makeScriptNum(v []byte, requireMinimal bool, scriptNumLen int) (scriptNum, error) {
+	// Interpreting data requires that it is not larger than
+	// the maximum scriptnum length or the passed scriptNumLen value.
+	if scriptNumLen > maxScriptNumLen || len(v) > scriptNumLen {
```

Comment:
The initial check against 9 can't be here.  While I agree with you that anything large than that number would cause an overflow and therefore should be invalidated, it can't be because that would disagree with the consensus rules.